cmd/lotus/daemon.go

```diff
@@ -232,6 +238,27 @@ var DaemonCmd = &cli.Command{
 
 		shutdownChan := make(chan struct{})
 
+		// If the daemon is started in "lite mode", replace the StateManager
+		// with a thin client to a gateway server
+		liteMode := node.Options()
+		if cctx.Bool("lite") {
+			gapi, closer, err := lcli.GetGatewayAPI(cctx)
+			if err != nil {
+				return err
+			}
+
+			createRPCStateMgr := func(lc fx.Lifecycle) *modules.RPCStateManager {
+				lc.Append(fx.Hook{
+					OnStop: func(ctx context.Context) error {
+						closer()
+						return nil
+					},
+				})
+				return modules.NewRPCStateManager(gapi)
+			}
+			liteMode = node.Override(new(stmgr.StateManagerAPI), createRPCStateMgr)
```

Comment:
I'd move `createRPCStateMgr` to a  function in `node/modules/lite.go`, and only set the repo type + put gateway API into DI (a bit like how lotus-miner puts FullAPI into its DI container)

Reply:
(We'll probably have quite a few of those functions)